WooCommerce/Classes/Authentication/Jetpack Setup/Native Jetpack Setup/LoginJetpackSetupViewModel.swift

```diff
@@ -8,15 +9,64 @@ final class LoginJetpackSetupViewModel: ObservableObject {
     /// Whether Jetpack is installed and activated and only connection needs to be handled.
     let connectionOnly: Bool
     private let stores: StoresManager
-    private let storeNavigationHandler: () -> Void
+    private let storeNavigationHandler: (_ connectedEmail: String?) -> Void
 
     let setupSteps: [JetpackInstallStep]
-    let title: String
+
+    /// Title to be displayed on the Jetpack setup view
+    var title: String {
+        let step = currentSetupStep ?? .installation
+        if setupFailed, let errorTitle = step.errorTitle {
+            return errorTitle
+        }
+        return connectionOnly ? Localization.connectingJetpack : Localization.installingJetpack
+    }
+
+    var shouldShowInitialLoadingIndicator: Bool {
+        currentSetupStep == nil && setupFailed == false
+    }
+
+    var shouldShowSetupSteps: Bool {
+        currentSetupStep != nil && setupFailed == false
+    }
+
+    var shouldShowGoToStoreButton: Bool {
+        currentSetupStep == .done && setupFailed == false
+    }
+
+    var tryAgainButtonTitle: String {
+        let step = currentSetupStep ?? .installation
+        return step.tryAgainButtonTitle ?? ""
+    }
 
     @Published private(set) var currentSetupStep: JetpackInstallStep?
     @Published private(set) var currentConnectionStep: ConnectionStep = .pending
     @Published private(set) var jetpackConnectionURL: URL?
     @Published var shouldPresentWebView = false
+    @Published var jetpackConnectionInterrupted = false
+
+    /// Whether the setup failed. This will be observed by `LoginJetpackSetupView` to present error modal.
+    @Published private(set) var setupFailed: Bool = false
+    @Published private(set) var setupErrorMessage: String = ""
+    @Published private(set) var setupErrorSuggestion: String = ""
+    @Published private(set) var errorCode: Int?
```

Comment:
Nit: If all of this all related, would it make sense to combine these into a `Struct`?

Reply:
Good point! Updated in f6037222df9e2a65eaf670a1b89720f195608946.